pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/InMemoryDelayedDeliveryTracker.java

```diff
@@ -111,8 +111,10 @@ public boolean addMessage(long ledgerId, long entryId, long deliverAt) {
         }
 
 
-        priorityQueue.add(deliverAt, ledgerId, entryId);
-        updateTimer();
+        synchronized (this) {
```

Comment:
I think it is not thread-safe when only lock this scope because `highestDeliveryTimeTracked` and `messagesHaveFixedDelay` will still be modified concurrently.

Reply:
what do you think add `synchronized` to InMemoryDelayedDeliveryTracker#addMessage.
 
and i have a question: InMemoryDelayedDeliveryTracker#getScheduledMessages is also need a `synchronized` ?

Reply:
> what do you think add synchronized to InMemoryDelayedDeliveryTracker#addMessage.

Yes, I think can add `synchronized` to InMemoryDelayedDeliveryTracker#addMessage or add `synchronized` to 
`PersistentDispatcherMultipleConsumers`

> i have a question: InMemoryDelayedDeliveryTracker#getScheduledMessages is also need a synchronized ?

I think`getScheduledMessages ` already add `synchronized ` on `PersistentDispatcherMultipleConsumers #getMessagesToReplayNow`:
https://github.com/apache/pulsar/blob/825b68db7bed1c79af4b7b69b48bee76ebe75af5/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java#L446-L457

Reply:
add a new commit, please review again